drivers/i2c/i2c_ll_stm32.h

```diff
@@ -60,15 +60,12 @@ struct i2c_stm32_data {
 	struct {
 #ifdef CONFIG_I2C_STM32_V1
 		unsigned int is_restart;
-		unsigned int flags;
 #endif
 		unsigned int is_write;
 		unsigned int is_arlo;
 		unsigned int is_nack;
 		unsigned int is_err;
-		struct i2c_msg *msg;
-		unsigned int len;
-		uint8_t *buf;
+		struct i2c_msg msg;
```

Comment:
We already duplicated the i2c_msg message when calling stm32_i2c_transaction() from i2c_stm32_transfer(). Why can't we use msg as a pointer and all flags and buffers within it?

Reply:
That would work for synchronous transfers, but for asynchronous transfers the copy of the `i2c_msg` created in the transfer function does not live long enough. If `current` just stored a pointer to that message, then as soon as the first write/read returned the memory allocated for the message would be released and the pointer would become invalid for the next write/read of the message.